weightedGraphs/networkDelayTime.py

```diff
@@ -0,0 +1,42 @@
+import heapq
+import collections
+
+'''
+  Assumptions:
+    N will be in the range [1, 100].
+    K will be in the range [1, N].
+    The length of times will be in the range [1, 6000].
+    All edges times[i] = (u, v, w) will have 1 <= u, v <= N and 1 <= w <= 100.
+
+    :type times: List[List[int]]
+    :type N: int
+    :type K: int
+    :rtype: int
+'''
+def network_delay_time(times, N, K):
```

Comment:
This was solved using Dijkstra’s algorithm, but I wonder if A* or greedy best-first could be used effectively? The solution is dependent on the assumption that returning the distance of the node furthest from the starting node is always correct. That said, could we use a priority queue configured to return the largest distance value instead of the smallest to arrive at our answer more quickly?